internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,335 @@
+	return revisionHash, nil
+}
+
+// parseCoverageSummary gets the branch coverage and line coverage from a coverage report summary.
+func parseCoverageSummary(content []byte) (map[string]float64, map[string]float64, error) {
+	var summary CoverageSummary
+	err := json.Unmarshal(content, &summary)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unmarshal: %v", err)
+	}
+	// Return branch coverage and line coverage using the coverage summary structure.
+	return summary.Data[0].Totals["branches"], summary.Data[0].Totals["lines"], nil
+}
+
+// getLogs gets the log-files list of a fuzz-target on a given day.
+func getLogs(date string, projectName string, fuzzTarget string, fuzzEngine string, sanitizer string) (*storage.BucketHandle, []string) {
+	// logsBucket is the ClusterFuzz Google Cloud Storage containing the fuzzers logs for a given project.
+	logsBucket := fmt.Sprintf("%s-logs.clusterfuzz-external.appspot.com", projectName)
+	bucket, err := getBucket(logsBucket)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fuzzengine := strings.ToLower(fuzzEngine)
+	// relativePath is the relative path in the logsBucket where the logs of a given fuzz-target on a given day are saved.
+	relativePath := fmt.Sprintf("%s_%s_%s/%s_%s_%s/%s", fuzzEngine, projectName, fuzzTarget, fuzzengine, sanitizer, projectName, date)
+	blobs, err := listBlobs(bucket, relativePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return bucket, blobs
+}
+
+// getFuzzEffortFromFile gets the fuzzingEffort from a single log file.
+func getFuzzEffortFromFile(reader io.Reader, revisionHash string, projectName string) (int, float64) {
+	var revisionHashInLog string
+	var err error
+	var timeFuzz float64
+	var numTests int
+	caser := cases.Title(language.English)
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		// get the revisionHash mentioned in the logfile.
+		if strings.Contains(scanner.Text(), caser.String(projectName)) {
+			revisionHashInLog = strings.Split(scanner.Text(), " ")[1]
+		}
+		// get the fuzzing time in seconds
+		if strings.Contains(scanner.Text(), "Time ran:") {
+			timeFuzz, err = strconv.ParseFloat(strings.Split(scanner.Text(), " ")[2], 32)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		// get the number of fuzz-tests
+		if strings.Contains(scanner.Text(), "stat::number_of_executed_units") {
+			numTests, err = strconv.Atoi(strings.Split(scanner.Text(), " ")[1])
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	// return the fuzzing efforts if the revisionHash in the logfile
+	// is the same as the revisionHash we are considering (it is
+	// possible to have logs for multiple revisions a given day)
+	if revisionHashInLog == revisionHash {
+		return numTests, timeFuzz
+	}
+	// the fuzzing effort does not count if the revisionHash in the
+	// logfile is different from the revisionHash we are considering.
+	return 0, 0.0
```

Comment:
Does it make sense to declare a type and return `nil` in this case? I think the type would be similar to `FuzzEffortSpec` that is now removed, but internal to the scraper :)

Reply:
We want `0` in this case and not `nil` because it is normal to have other log files that are not related to the revision of the source code at a given date. We just want to ignore them (it's not an error).

I update the code to make the idea more clear :))

Reply:
Sure, in that case you can return a default object instead of `nil`. `nil` does not necessarily represent an error.